Userland/Utilities/sort.cpp

```diff
@@ -7,34 +7,60 @@
 #include <AK/QuickSort.h>
 #include <AK/String.h>
 #include <AK/Vector.h>
+#include <LibCore/ArgsParser.h>
+#include <LibCore/File.h>
 #include <errno.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
 #include <unistd.h>
 
-int main([[maybe_unused]] int argc, [[maybe_unused]] char** argv)
+int main(int argc, char** argv)
 {
-    if (pledge("stdio", nullptr) > 0) {
+    if (pledge("stdio rpath", nullptr) > 0) {
         perror("pledge");
         return 1;
     }
 
+    Vector<const char*> files;
     Vector<String> lines;
 
-    for (;;) {
-        char* buffer = nullptr;
-        ssize_t buflen = 0;
-        size_t n;
-        errno = 0;
-        buflen = getline(&buffer, &n, stdin);
-        if (buflen == -1 && errno != 0) {
-            perror("getline");
-            exit(1);
+    Core::ArgsParser args_parser;
+    args_parser.add_positional_argument(files, "File(s) to process", "file", Core::ArgsParser::Required::No);
+    args_parser.parse(argc, argv);
+
+    auto handle_file = [&](StringView filename) -> bool {
+        auto file = Core::File::construct(filename);
+        if (!file->open(Core::OpenMode::ReadOnly)) {
+            warnln("Failed to open {}: {}", filename, file->error_string());
+            return false;
+        }
+
+        while (file->can_read_line())
+            lines.append(file->read_line());
+        return true;
+    };
+
+    if (!files.size()) {
```

Comment:
I am not sure if it even works, but what about using `File::standard_input()`, so that you can make use of the same logic above for stdin.

Reply:
I made this change. Although, I think I may have found a bug in doing so. You can correct me if I'm doing something wrong. Basically, if I do something like `cat file1 file2 | sort`, there is a chance that a I get an error that looks something like this:

```
Shell: Job 1 (cat file1 file2) Broken pipe
```

This error doesn't happen every time, however. The reason I think it is a bug is because I have tested a similar thing with some of the other utilities using `File::standard_input()` and they also report the same error.

Reply:
`File::standard_input()` does not seem to be thread safe, that could be the issue.

Reply:
Oh alright. Should I revert or just roll with this change?